pkg/apis/serving/v1beta1/component.go

```diff
@@ -248,6 +270,30 @@ type MetricSource struct {
 	// For namespaced query
 	// +optional
 	Namespace string `json:"namespace,omitempty"`
+	// OperationOverTime specifies the operation to aggregate the metrics over time
+	// possible values are last_one, avg, max, min, rate, count. Default is 'last_one'.
+	// +optional
+	OperationOverTime string `json:"operationOverTime,omitempty"`
+}
+
+type PodsMetricSource struct {
+	// MetricsBackend defines the scaling metric type watched by autoscaler
+	// possible values are prometheus, graphite, opentelemetry.
+	// +optional
+	Backend *PodsMetricsBackend `json:"backend,omitempty"`
+	// Address of MetricsBackend server.
+	// +optional
+	ServerAddress string `json:"serverAddress,omitempty"`
+	// MetricName is the name of the metric in the backend
+	// +optional
+	MetricName string `json:"metricName,omitempty"`
```

Comment:
the query may have multiple metrics, also if this is left not specified then it filters all the metrics.